src/Math/tests/WrenchTest.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * @file WrenchTest.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+// Catch2
+#include <catch2/catch.hpp>
+
+#include <BipedalLocomotion/Math/Wrench.h>
+
+#include <manif/SE3.h>
+#include <manif/SO3.h>
+
+using namespace BipedalLocomotion::Math;
+
+TEST_CASE("Wrench test")
+{
+    const Wrenchd wrench = Wrenchd::Random();
+
+    SECTION("Get force and torque")
+    {
+        REQUIRE(wrench.head<3>().cwiseEqual(wrench.force()).all());
+        REQUIRE(wrench.tail<3>().cwiseEqual(wrench.torque()).all());
+    }
+
+    SECTION("set force and torque")
+    {
+        Wrenchd newWrench;
+        newWrench.force() = wrench.force();
+        newWrench.torque() = wrench.torque();
+
+        REQUIRE(newWrench.force().cwiseEqual(wrench.force()).all());
+        REQUIRE(newWrench.torque().cwiseEqual(wrench.torque()).all());
+    }
+
+    SECTION("Apply transform")
+    {
+        const manif::SE3d transform = manif::SE3d::Random();
+        const Wrench transformedWrench = transform * wrench;
+
+        // Get the adjoint matrix for spatial force vector
+        manif::SE3d::Jacobian adjointMatrix = transform.adj();
+        adjointMatrix.bottomLeftCorner<3, 3>() = adjointMatrix.topRightCorner<3, 3>();
+        adjointMatrix.topRightCorner<3, 3>().setZero();
```

Comment:
Possibly, could you just add a comment that this is a conversion from `AdjointTransform` to `AdjointWrenchTransform`. When someone else reads this test, it will be quick to understand the operation.

Reply:
https://github.com/dic-iit/bipedal-locomotion-framework/pull/279/commits/9022f13b4c8baa51adb04ed04878f36a946d8c9a